### PR TITLE
Fixes based on comprehensive code review

### PR DESCRIPTION
**Pull request by poldrack**
This PR includes changes based on an AI-driven code review and refactor. Below is Copilot's summary. Note that the actual production change itself is small:  fracridge/fracridge.py +87/−36 and fracridge/_linalg.py +3/−3.   A notebook (examples/diagnose_intercept_and_svd_inaccuracy.ipynb) is also included to demonstrate the two major issues (intercept handling and SVD computation), and the test suite was also expanded significantly.  Added Python 3.14 to testing matrix and confirmed that it runs.

(sorry for bundling this all together into a big PR!)

## Copilot summary

This pull request introduces significant improvements to the numerical accuracy, reliability, and developer experience of the `fracridge` package. The most important changes include switching the core SVD computation to operate directly on the design matrix for improved numerical stability, fixing intercept handling in cross-validated regression, and greatly expanding the test suite to cover both legacy and new behaviors. Additionally, developer tooling and documentation have been enhanced to support these changes.

**Core algorithm and numerical accuracy:**

* The SVD computation in `fracridge` now decomposes the design matrix `X` directly, rather than the normal matrix `X.T @ X`, improving numerical stability and ensuring accurate coefficients even for nearly collinear designs. This change is not configurable and is documented as a deliberate trade-off of speed for accuracy. [[1]](diffhunk://#diff-5444e14b636b4fbf9929c56293db54bed24b0870b7aff7966c3e7bc02126651cL45-R90) [[2]](diffhunk://#diff-5444e14b636b4fbf9929c56293db54bed24b0870b7aff7966c3e7bc02126651cL159-R181) [[3]](diffhunk://#diff-4b46e99840eeda55b0a1d8078707997092936e42dcc8804f25f165b6abe2a90fR69-R88)
* Helper functions for SVD selection and shrinkage computation have been refactored for clarity and correctness, ensuring inactive singular values are handled safely and avoiding division by zero. [[1]](diffhunk://#diff-5444e14b636b4fbf9929c56293db54bed24b0870b7aff7966c3e7bc02126651cL45-R90) [[2]](diffhunk://#diff-5444e14b636b4fbf9929c56293db54bed24b0870b7aff7966c3e7bc02126651cL185-R204) [[3]](diffhunk://#diff-5444e14b636b4fbf9929c56293db54bed24b0870b7aff7966c3e7bc02126651cL212-R231)

**Bug fixes and reliability:**

* Fixed a bug in `FracRidgeRegressorCV` where the constructor incorrectly set `self` as the base class's `fracs` argument, and corrected intercept handling so that predictions and scoring now properly account for the fitted intercept in both single and multi-target cases. [[1]](diffhunk://#diff-5444e14b636b4fbf9929c56293db54bed24b0870b7aff7966c3e7bc02126651cL436-R459) [[2]](diffhunk://#diff-5444e14b636b4fbf9929c56293db54bed24b0870b7aff7966c3e7bc02126651cL448-R471) [[3]](diffhunk://#diff-46a37724e80838cdfbef5cc2f508524e7df9d13b787de3ec8e15f8f99b4ca645R1-R91)

**Testing and regression coverage:**

* Added comprehensive new tests for cross-validation, intercept recovery, and multi-target regression, as well as a regression test that executes and verifies the accompanying diagnostic notebook. These tests ensure that both the new direct-SVD path and legacy behaviors are validated. [[1]](diffhunk://#diff-46a37724e80838cdfbef5cc2f508524e7df9d13b787de3ec8e15f8f99b4ca645R1-R91) [[2]](diffhunk://#diff-4c7f42ade8701e753685df9b3bb0edbcf146daec8bf62ee1e9d8354ea7f294b6R1-R85) [[3]](diffhunk://#diff-eb9c211deb83f6088cd2027b8512c8148216b7d37fc8e368e0de8447d4e5fccbR8-R55) [[4]](diffhunk://#diff-eb9c211deb83f6088cd2027b8512c8148216b7d37fc8e368e0de8447d4e5fccbR66-R69) [[5]](diffhunk://#diff-eb9c211deb83f6088cd2027b8512c8148216b7d37fc8e368e0de8447d4e5fccbR82-R85) [[6]](diffhunk://#diff-eb9c211deb83f6088cd2027b8512c8148216b7d37fc8e368e0de8447d4e5fccbR98-R101) [[7]](diffhunk://#diff-eb9c211deb83f6088cd2027b8512c8148216b7d37fc8e368e0de8447d4e5fccbR113-R115) [[8]](diffhunk://#diff-eb9c211deb83f6088cd2027b8512c8148216b7d37fc8e368e0de8447d4e5fccbR129-R132)

**Developer tooling and documentation:**

* Added a `Makefile` with developer entry points for running, checking, and linting the diagnostic notebook, as well as running the full test suite, improving the developer workflow.
* Updated the user guide to explain the rationale for the slower solver and the trade-off between speed and accuracy.
* Added Python 3.14 as a testing target in the Github Actions workflow

**Internal code and dependency cleanup:**

* Minor fixes to imports and function calls for LAPACK routines, ensuring correct usage of `get_lapack_funcs` and improving compatibility. [[1]](diffhunk://#diff-01f3c2e62eae3e51465b971ce6e498968cd9a242fb0fa27be33a9cfb0ffbf3e4L39-R39) [[2]](diffhunk://#diff-01f3c2e62eae3e51465b971ce6e498968cd9a242fb0fa27be33a9cfb0ffbf3e4L102-R104)

These changes collectively make `fracridge` more robust, accurate, and maintainable, with a strong focus on correctness and reproducibility.

### CODE REVIEW

File: fracridge/tests/test_fracridge.py [hunk 1]
```diff
@@ -5,32 +5,54 @@
...
     fracridge(X, y, fracs=fracs, jit=jit)
 
 
 @pytest.mark.parametrize("nn, pp", [(1000, 10), (10, 100), (284, 50)])
 @pytest.mark.parametrize("bb", [(1), (2), (1000)])
 @pytest.mark.parametrize("jit", [True, False])
 def test_benchmark_fracridge(nn, pp, bb, jit, benchmark):
+    """Records timings only; it asserts nothing and so cannot fail on speed.
+    Useful for watching the cost of the direct-SVD solve, which is markedly
+    slower on tall designs than the old normal-equations path.
```
Comment (arokem) [inline fracridge/tests/test_fracridge.py]: "old" here depends on knowing the history of the project before this PR, and may not be useful as documentation in the future. Might need to unpack this.

File: fracridge/tests/test_fracridge.py
Comment (arokem) [inline fracridge/tests/test_fracridge.py]: Wasn't this test supposed to fail prior to these changes? Did atol improve with these changes?

Comment (arokem) [reply fracridge/tests/test_fracridge.py]: Or maybe this was not the right setup? That is, `make_data` does not create data that is ill-conditioned enough?

File: fracridge/tests/test_cross_validation.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
+import numpy as np
+import pytest
+from sklearn.model_selection import KFold
+
+from fracridge.fracridge import FracRidgeRegressorCV
+
+INTERCEPT = 3.0
+SLOPE = 2.0
+SECOND_INTERCEPT = -1.5
+SECOND_SLOPE = 0.5
```
Comment (arokem) [inline fracridge/tests/test_cross_validation.py]: Would rather have these as inputs to `mark_parameterize` with a few different values to test. E.g., zero intercept, negative slope, etc.

File: fracridge/tests/test_diagnostic_notebook.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
+import json
```
Comment (arokem) [inline fracridge/tests/test_diagnostic_notebook.py]: I think this file should be removed before merging

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,39 @@
+# Developer entry points for fracridge.
```
Comment (arokem) [inline Makefile]: I don't think we need this

File: examples/diagnose_intercept_and_svd_inaccuracy.ipynb [hunk 5]
```diff
@@ -0,0 +1,793 @@
+{
+ "cells": [
```
Comment (arokem) [inline examples/diagnose_intercept_and_svd_inaccuracy.ipynb]: I believe this file should be removed before merging this PR

**Review by arokem — COMMENTED** [review]
Thanks for these fixes and apologies that it took so long to review them. I believe the diagnostic notebook and related files should be removed before merging. Other than that, my comments are minor and/or a clarification question.

**Comment by poldrack** [comment]
thanks! I'm teaching pretty heavily this quarter so it might take me a couple of weeks to be able to dig back into this
